File: app/src/main/java/com/msk/produtosperigosos/info/SobreApp.java
package com.msk.produtosperigosos.info;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager.NameNotFoundException;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.msk.produtosperigosos.R;

public class SobreApp extends Fragment {

    private View tela;
    private PackageInfo pinfo;
    private TextView versao;
    private String nr;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        tela = inflater.inflate(R.layout.sobre, container, false);

        versao = (TextView) tela.findViewById(R.id.tvVersaoApp);

        try {
            pinfo = getActivity().getPackageManager().getPackageInfo(getActivity().getPackageName(), 0);
        } catch (NameNotFoundException e) {
            e.printStackTrace();
        }
        nr = pinfo.versionName;

        versao.setText(getResources().getString(R.string.desenvolvedor, nr));

        return tela;
    }


}

File: src/main/java/com/msk/produtosperigosos/telas/SobreApp.kt
package com.msk.produtosperigosos.telas

import android.content.pm.PackageManager
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.msk.produtosperigosos.R

class SobreApp : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val view = inflater.inflate(R.layout.sobre, container, false)
        val versionTextView = view.findViewById<TextView>(R.id.tvVersaoApp)

        try {
            val versionName = requireActivity().packageManager
                .getPackageInfo(requireActivity().packageName, 0).versionName
            versionTextView.text = getString(R.string.desenvolvedor, versionName)
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
            versionTextView.text = getString(R.string.desenvolvedor, "N/A")
        }

        return view
    }
}